### PR TITLE
include changes upto v2.5.0.pre129

### PR DESCRIPTION
**Pull request by priyavadan**
Stephen,

Have included some changes to the Core and brought it up to speed with v2.5.0.pre129. Please let me know.

Thanks,
Priyavadan

### CODE REVIEW

**Comment by priyavadan** [comment]
Found an issue, need to fix and re-open

File: lib/origen/commands/fetch.rb [hunk 1]
```diff
@@ -12,8 +12,18 @@
 Usage: origen fetch [PLUGIN_NAME] [options]
 END
   opts.on('-o', '--outputdir', 'User defined destination directory of the plugin workspace') {  options[:debugger] = true }
+  opts.on('-v', '--version', 'User requested plugin/gem version') {  options[:debugger] = true }
   opts.on('-h', '--help', 'Show this message') { puts opts; exit }
 end
+
+# always the first argument
```
Comment (ginty) [inline lib/origen/commands/fetch.rb]: This whole section should be removed to use the option parser above

File: lib/origen/commands/fetch.rb [hunk 2]
```diff
@@ -25,13 +35,13 @@
 valid_name = false
 
 # If requested plugin is origen_core or origen, then run special case, else search through all available plugins
-if plugin_name.downcase == 'origen_core' || plugin_name.downcase == 'origen'
-  plugin_name = 'origen'
+if @plugin_name.downcase == 'origen_core' || @plugin_name.downcase == 'origen'
```
Comment (ginty) [inline lib/origen/commands/fetch.rb]: No need to turn these into instance variables

File: lib/origen/commands/fetch.rb [hunk 3]
```diff
@@ -40,24 +50,24 @@
...
   exit 1
 end
 
 # If user didn't specify directory then configure for default directory
 unless dir
-  dir = "./#{plugin_name}"
+  dir = "./#{@plugin_name}"
   puts "You did not specify a workspace directory for #{plugin_name} plugin, it will therefore be created at"
-  puts "#{dir}"
+  puts "#{@dir}"
```
Comment (ginty) [inline lib/origen/commands/fetch.rb]: Should not be an instance var either

File: lib/origen/revision_control/design_sync.rb [hunk 4]
```diff
@@ -191,16 +191,38 @@ def root
...
           root = local
+          # Create two arrays, one to store the parents and the other to corresponding vaults.
+          temp_parent_array = Array.new(0)
+          temp_parent_vault_array = Array.new(0)
           resolved = false
           vault = dssc("url vault #{root}").first
           until resolved || root.root?
             parent = root.parent
+            # push the current parent into the parent array
+            temp_parent_array.push("#{parent}")
```
Comment (ginty) [inline lib/origen/revision_control/design_sync.rb]: Could you change "#{parent}" and similar occurrences to parent.to_s

**Comment by ginty** [comment]
Hi @priyavadan, thanks a lot for doing this!
I know in most cases it is not your code, however could you please make the suggested updates.
Thanks!

**Comment by priyavadan** [comment]
Hi @ginty , I have updated some files, please take a look when you get a chance. 

Thanks!
Priyavadan

**Comment by ginty** [comment]
Looks good to me, thanks!

**Pull request merged**